roles/kubernetes/preinstall/tasks/0020-verify-settings.yml

```diff
@@ -293,6 +293,12 @@
     msg: "kata_containers_enabled support only for containerd and crio-o. See https://github.com/kata-containers/documentation/blob/1.11.4/how-to/run-kata-with-k8s.md#install-a-cri-implementation for details"
   when: kata_containers_enabled
 
+- name: Stop if gvisor_enabled is enabled when container_manager is not containerd
+  assert:
+    that: container_manager == 'containerd'
+    msg: "gvisor_enabled support only compatible with containerd. See https://github.com/kubernetes-sigs/kubespray/issues/7650 for details"
+  when: gvisor_enabled
+
```

Comment:
The doc mentions cri-o too. Which one is correct?

Reply:
I updated the docs. As per @ianlewis' comment, cri-o is not currently usable though it does support setting the `RuntimeClass`; because of this I restricted the validation to `containerd`. Once gVisor adds support for cri-o, we can change this validation and the code to support it is already in this PR.

Reply:
Thanks @cristicalin for this PR!